webapp/src/components/AssessmentMetadataBar.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  Avatar,
+  Divider,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Tooltip,
+  Typography,
+  Switch,
+} from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
+import InventoryIcon from '@mui/icons-material/Inventory';
+import TimerIcon from '@mui/icons-material/Timer';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import LightbulbIcon from '@mui/icons-material/Lightbulb';
+import { blue, grey } from '@mui/material/colors';
+
+import {
+  Assessment,
+  SubmissionStatus,
+  AssessmentType,
+  AssessmentSubmission,
+} from '../assets/data';
+import { formatDateTime } from '../helpers/dateTime';
+
+interface AssessmentMetadataBarProps {
+  assessment: Assessment;
+  submission: AssessmentSubmission;
+}
+
+const AssessmentMetadataBar = ({
+  assessment,
+  submission,
+}: AssessmentMetadataBarProps) => {
+  const dueTime = new Date(assessment?.dueDate!);
+  //TODO: use the opened date from the submission
+  const [openedTime] = useState(new Date());
+  const [secondsRemaining, setSecondsRemaining] = useState(
+    assessment?.testDuration! * 60
+  );
+
+  const requestRef = useRef<number>();
+  const previousTimeRef = useRef<number>();
+
+  const animate = (time: number) => {
+    if (previousTimeRef.current !== undefined) {
+      if (
+        new Date().getTime() > dueTime.getTime() ||
+        openedTime.getTime() + assessment?.testDuration! * 60 * 1000 <
+          new Date().getTime()
+      ) {
+        setSecondsRemaining(0);
+      } else if (
+        new Date().getTime() + assessment?.testDuration! * 60 * 1000 >
+        dueTime.getTime()
+      ) {
+        setSecondsRemaining(
+          Math.round((dueTime.getTime() - new Date().getTime()) / 1000)
+        );
+      } else {
+        setSecondsRemaining(
+          Math.round(
+            (openedTime.getTime() +
+              assessment?.testDuration! * 60 * 1000 -
+              new Date().getTime()) /
+              1000
+          )
+        );
+      }
+    }
+    previousTimeRef.current = time;
+    requestRef.current = requestAnimationFrame(animate);
+  };
+
+  useEffect(() => {
+    requestRef.current = requestAnimationFrame(animate);
+    return () => {
+      if (requestRef.current) cancelAnimationFrame(requestRef.current);
+    };
+  });
+
+  const [showTimer, setShowTimer] = React.useState(true);
+
+  const handleSetShowTimer = (showTimer: boolean) => () => {
+    setShowTimer(showTimer);
+  };
+
+  return (
+    <>
+      <List
+        sx={{
+          width: '100%',
+          maxWidth: 360,
+        }}
+      >
+        <ListItem>
+          <ListItemAvatar>
+            <Avatar
+              sx={{
+                bgcolor: `${
+                  submission.state === SubmissionStatus.Opened
+                    ? blue[600]
```

Comment:
Instead of using `blue[600]` and `grey[400]` everywhere in our app, may I recommend saving these values to some variables called `enabledBgColor` and `disabledBgColor`? That way if we want to update / change the colors later we only have to edit the variables and not have to find/replace.